xds/internal/xdsclient/client.go

```diff
@@ -24,13 +24,49 @@ import (
 	"fmt"
 	"time"
 
+	"google.golang.org/grpc"
 	"google.golang.org/grpc/internal/grpclog"
 	"google.golang.org/grpc/internal/grpcsync"
 	"google.golang.org/grpc/xds/internal/xdsclient/bootstrap"
 	"google.golang.org/grpc/xds/internal/xdsclient/pubsub"
 	"google.golang.org/grpc/xds/internal/xdsclient/xdsresource"
 )
 
+var controllerDialOpts []grpc.DialOption
+
+// SetControllerDialOpts sets extra dial options for the xds controller
+// which connects to the xDS server.
+// Must be called before client is created(ex: in init()), or the xds Client
+// might have been created before reading the set dial options.
```

Comment:
Even an `init` might not be able to call this early enough.  (I.e. if something is imported that creates the client, that will happen before your `init`.)  I thought we were going to resolve this by doing this in an `init` that we add to _this same package_ (in our repo, not in OSS).